content/blog/2017/10/vmware-disks.md

```diff
@@ -0,0 +1,391 @@
+
+## Blank Controller
+
+Example of a blank controller:
+
+```python
+from pyVmomi import vim
+
+controller = vim.vm.device.VirtualSCSIController()
+
+output:
+  (vim.vm.device.VirtualSCSIController) {
+     dynamicType = <unset>,
+     dynamicProperty = (vmodl.DynamicProperty) [],
+     key = 0,
+     deviceInfo = <unset>,
+     backing = <unset>,
+     connectable = <unset>,
+     slotInfo = <unset>,
+     controllerKey = <unset>,
+     unitNumber = <unset>,
+     busNumber = 0,
+     device = (int) [],
+     hotAddRemove = <unset>,
+     sharedBus = <unset>,
+     scsiCtlrUnitNumber = <unset>
+  }
+```
+
+This is very similar to the Disks configuration in that most of the options do not need to be filled out. However, some do, something to keep in mind is that while the busNumber is initialized at 0 this is already taken by the first controller on the VM. This busNumber can be 0 to 3 since there can only be 4 controllers per VM.
+
+## Example Controller
+
+Here is an example of a controller that has been filled in to be added.
+
+```python
+from pyVmomi import vim
+
+controller = vim.vm.device.VirtualSCSIController()
+controller.sharedBus = 'noSharing'
+controller.busNumber = 1
+
+output:
+  (vim.vm.device.VirtualSCSIController) {
+     dynamicType = <unset>,
+     dynamicProperty = (vmodl.DynamicProperty) [],
+     key = 0,
+     deviceInfo = <unset>,
+     backing = <unset>,
+     connectable = <unset>,
+     slotInfo = <unset>,
+     controllerKey = <unset>,
+     unitNumber = <unset>,
+     busNumber = 1,
+     device = (int) [],
+     hotAddRemove = <unset>,
+     sharedBus = 'noSharing',
+     scsiCtlrUnitNumber = <unset>
+  }
+```
+
+Once you have your configuration for the addition that you want to make to the vmWare VM you need to generate a configuration specification to be passed into a reconfigure VM task to send to vmWare. This accepts a list of configurationss that can be done at the same time if you wish. Keep in mind that if you need to generate a new controller to add a disk then that will need to be done before so that you can get auto generated information unless you specify it.
+
+# Reconfigure VM Task
+
+Example of how to add a disk using the VM object from above:
+
+```python
+from pyVmomi import vim
+
+# Generate basic Disk information
+disk = vim.vm.device.VirtualDisk()
+disk.backing = vim.vm.device.VirtualDisk.FlatVer2BackingInfo()
+disk.backing.diskMode = "persistent"
+disk.capacityInKB = 1024
+disk.controllerKey = 1
+disk.backing.datastore = vim.Datastore("datastore")
+disk.unitNumber = 1
+
+# Generate virtual device Specification
+disk_spec = vim.vm.device.VirtualDeviceSpec()
+disk_spec.fileOperation = "create"
+disk_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
+
+# Add disk to the device spec
+disk_spec.device = disk
+
+config_spec = vim.vm.ConfigSpec()
+config_spec.deviceChange = [disk_spec]
+
```

Comment:
Make a comment here that calling this function applies the new configuration (actually does the work)